pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -197,6 +197,13 @@ func (r *Reconciler) reconcileMachineWithCloudState(failedCondition *v1beta1.GCP
 			Type:    corev1.NodeInternalDNS,
 			Address: fmt.Sprintf("%s.c.%s.internal", r.machine.Name, r.projectID),
 		})
+		// Add the machine's name as a known NodeInternalDNS because GCP platform
+		// provides search paths to resolve those.
+		// https://cloud.google.com/compute/docs/internal-dns#resolv.conf
+		nodeAddresses = append(nodeAddresses, corev1.NodeAddress{
+			Type:    corev1.NodeInternalDNS,
```

Comment:
do we need to also include this as `NodeHostName`?

Reply:
Not really, this VM name is resolvable internal DNS name for the machine on GCP. We can't say what the hostname will be.